### PR TITLE
Store delete - 2nd attempt

### PR DESCRIPTION
**Pull request by beldmit**
An alternative to #21869

The objects to be deleted are identified by URI

- [X] documentation is added or updated
- [X] tests are added or updated

### CODE REVIEW

**Comment by beldmit** [comment]
@levitte I didn't find how to make the code less clumsy when passing ui_method/ui_data (the PW callback struct is not an option, I think, and a dummy context is also not the best option).

The open question is whether we need to pass the params to the delete function.

**Comment by beldmit** [comment]
@simo5 could you please take a look?

**Comment by simo5** [comment]
You need to provide a password callback to the store, I need it to be able to delete objects form pkcs#11.

**Comment by beldmit** [comment]
@simo5 is ui_method/ui_data not enough?

**Comment by simo5** [comment]
I never dealt with it in the provider that I recall, the rest of the STORE API provides a pwd callback though, I would like consistency.

**Comment by beldmit** [comment]
@levitte what is the best way then? Dummy store context?

**Comment by beldmit** [comment]
@simo5 I inherited an approach from #20131. 
@levitte ready for review, I think.

File: include/openssl/core_dispatch.h [hunk 1]
```diff
@@ -951,6 +952,9 @@ OSSL_CORE_MAKE_FUNC(int, store_close, (void *loaderctx))
 OSSL_CORE_MAKE_FUNC(int, store_export_object,
                     (void *loaderctx, const void *objref, size_t objref_sz,
                      OSSL_CALLBACK *export_cb, void *export_cbarg))
+OSSL_CORE_MAKE_FUNC(int, store_delete, (void *provctx, const char *uri,
+                    const OSSL_PARAM params[],
+                    OSSL_PASSPHRASE_CALLBACK *pw_cb, void *pw_cbarg))
```
Comment (levitte) [inline include/openssl/core_dispatch.h:957]:
```suggestion
OSSL_CORE_MAKE_FUNC(int, store_delete,
                    (void *provctx, const char *uri, const OSSL_PARAM params[],
                     OSSL_PASSPHRASE_CALLBACK *pw_cb, void *pw_cbarg))
```

**Review by levitte — CHANGES_REQUESTED** [review]
One code style nit.  Otherwise, I like what I see

File: include/openssl/core_dispatch.h (hunk 1 above)
Comment (beldmit) [reply include/openssl/core_dispatch.h]: Fixed and pushed. Thanks!

**Review by levitte — APPROVED** [review]
This is clear, this is neat.  Yup!

**Comment by beldmit** [comment]
I kindly ask @openssl/committers for the 2nd approval

**Comment by levitte** [comment]
Actually, @openssl/otc.  I want another OTC pair of eyes on this

**Review by t8m — APPROVED** [review]

**Comment by beldmit** [comment]
Merged. Thanks for the reviews!

**Pull request closed without merging**